core/sr-primitives/src/generic/header.rs

```diff
@@ -26,10 +26,11 @@ use generic::Digest;
 #[cfg_attr(feature = "std", derive(Debug, Serialize))]
 #[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
 #[cfg_attr(feature = "std", serde(deny_unknown_fields))]
-pub struct Header<Number, Hash: HashT, DigestItem> {
+pub struct Header<Number: Copy + Into<u128>, Hash: HashT, DigestItem> {
```

Comment:
While it's unlikely that chains could have block number >= 2^128, that might be assuming that they all have a genesis with block 0 in the future.

I could see someone doing a clever "tree of chains" thing where some initial bits are used to encode a chain ID or similar.

Reply:
i think that as long as we require `As<u64>`, then we're good here. If a chain relied on having non-parent relationships on the same continuum of blocknumbers, then they're asking for trouble.